### ISSUE TITLE
[Question]: Can broker keys be used with market data clients?

### ISSUE THREAD

**Issue by builder173** [alpacahq/alpaca-py#331]
### Question form pre-submit checklist.

- [X] I have searched the existing issues to ensure there isn't already an issue about this question.
- [X] My question has to do with the Python SDK and isn't a general question about the API. (If it is please open your issue [here](https://github.com/alpacahq/Alpaca-API))
- [X] My question isn't about how to do a specific algorithm or asking for trade advice (answers to these are outside the scope of this repo).

### Question

The API reference [here](https://alpaca.markets/docs/api-references/market-data-api/stock-pricing-data/) indicates that the Broker API key can be used to access IEX market data. Does that hold for the python library as well?

I attempted to follow the historical stock price example [here](https://alpaca.markets/docs/python-sdk/market_data.html#market-data) using my broker api key/secret. 

```
alp_stock_client = StockHistoricalDataClient( 
api_key=os.getenv('ALPACA_API_KEY'), 
secret_key=os.getenv('ALPACA_API_SECRET'))
multisymbol_request_params = StockLatestQuoteRequest(symbol_or_symbols=["SPY", "GLD", "TLT"])
latest_multisymbol_quotes = alp_stock_client.get_stock_latest_quote(multisymbol_request_params)
gld_latest_ask_price = latest_multisymbol_quotes["GLD"].ask_price

```
However I'm getting this error:
HTTPError: 403 Client Error: Forbidden for url: https://data.alpaca.markets/v2/stocks/quotes/latest?symbols=SPY%2CGLD%2CTLT

If I add the url_override parameter with value 'https://data.sandbox.alpaca.markets' I get the same error.

I can setup a trading account too if necessary, just don't want to acquire another set of keys if the broker api keys are supposed to work here.

Thanks

**Comment by haxdds** [comment]
Yes it is possible to0 use your Broker API keys for Market Data. You need to make sure to set the `use_basic_auth` parameter to `False` within StockHistoricalData (https://github.com/alpacahq/alpaca-py/blob/527545dc6406569419220baa94dbb1d72cf55256/alpaca/data/historical/stock.py#L48C9-L48C38). This is unlike the Trading/Data API, Broker API uses basic auth.

**Comment by builder173** [comment]
@haxdds 
Thanks, I assume you meant I should set `use_basic_auth` to `True` since `False` is the default?

Although `use_basic_auth=True` didn't work on its own, it did work in combination with:
`url_override='https://data.sandbox.alpaca.markets'`
